lang/java/reef-wake/wake/src/test/java/org/apache/reef/wake/test/time/ClockTest.java

```diff
@@ -118,19 +124,22 @@ public void testMultipleCloseCalls() throws Exception {
     LoggingUtils.setLoggingLevel(Level.FINE);
 
     final int numThreads = 3;
+    final CountDownLatch eventCountLatch = new CountDownLatch(numThreads);
+
     final RuntimeClock clock = buildClock();
     new Thread(clock).start();
     final ThreadPoolStage<Alarm> stage = new ThreadPoolStage<>(new EventHandler<Alarm>() {
       @Override
       public void onNext(final Alarm value) {
         clock.close();
+        eventCountLatch.countDown();
       }
     }, numThreads);
 
     try {
       for (int i = 0; i < numThreads; ++i)
         stage.onNext(null);
-      Thread.sleep(1000);
+      eventCountLatch.await(10, TimeUnit.SECONDS);
```

Comment:
I'm concerned about the case when the CountDownLatch times out. For the current test code, it's indistinguishable between (a) all three `clock.close()` calls complete, and (b) one or more stall (e.g., from a deadlock). If we wrap `assertTrue` the test would fail on (b) because await returns false.